lisa/sut_orchestrator/azure/features.py

```diff
@@ -182,6 +185,191 @@ def is_over_nd(self) -> bool:
+            file="/etc/selinux/config",
+            sudo=True,
+        )
+
+        # Install OFED
+        mofed_version = "5.4-3.0.3.0"
+        if isinstance(node.os, Redhat):
+            os_class = "rhel"
+        else:
+            os_class = node.os.name.lower()
+
+        os_version = node.os.information.release.split(".")
+        mofed_folder = (
+            f"MLNX_OFED_LINUX-{mofed_version}-{os_class}"
+            f"{os_version[0]}."
+            f"{os_version[1]}-x86_64"
+        )
+        tarball_name = f"{mofed_folder}.tgz"
+        mlnx_ofed_download_url = (
+            f"https://partnerpipelineshare.blob.core.windows.net/ofed/{tarball_name}"
+        )
+
+        node.execute(
+            f"wget {mlnx_ofed_download_url}",
+            expected_exit_code=0,
+            expected_exit_code_failure_message="MLNX OFED Drivers not found at "
+            f"{mlnx_ofed_download_url}",
+        )
+        node.execute(
+            f"tar zxvf {tarball_name}",
+            expected_exit_code=0,
+            expected_exit_code_failure_message="SetupRDMA: failed to unzip "
+            f"{tarball_name}",
+        )
+
+        node.execute(
+            f"./{mofed_folder}/mlnxofedinstall --kernel {kernel} "
+            f"--kernel-sources /usr/src/kernels/{kernel} --add-kernel-support "
+            f"--skip-repo --skip-unsupported-devices-check --without-fw-update",
+            expected_exit_code=0,
+            expected_exit_code_failure_message="SetupRDMA: failed to install "
+            "MOFED Drivers",
+            sudo=True,
+        )
+        node.execute(
+            "/etc/init.d/openibd force-restart",
+            expected_exit_code=0,
+            expected_exit_code_failure_message="SetupRDMA: failed to restart driver",
+            sudo=True,
+        )
+
+        # Update waagent.conf
+        sed.substitute(
+            regexp="# OS.EnableRDMA=y",
+            replacement="OS.EnableRDMA=y",
+            file="/etc/waagent.conf",
+            sudo=True,
+        )
+        sed.substitute(
+            regexp="# AutoUpdate.Enabled=y",
+            replacement="AutoUpdate.Enabled=y",
+            file="/etc/waagent.conf",
+            sudo=True,
+        )
+
+        systemctl = node.tools[Systemctl]
+        systemctl.run(
+            parameters="daemon-reload",
+            expected_exit_code=0,
+            expected_exit_code_failure_message="SetupRDMA: failed to reload daemons",
+            sudo=True,
+        )
+        if isinstance(node.os, Ubuntu):
+            systemctl.restart_service("walinuxagent")
+        else:
+            systemctl.restart_service("waagent")
+
+        # Intall Intel MPI
+        wget = node.tools[Wget]
+        script_path = wget.get(
+            "https://partnerpipelineshare.blob.core.windows.net/mpi/"
+            "l_mpi_oneapi_p_2021.1.1.76_offline.sh"
+        )
+        node.execute(
+            f"bash {script_path} -s -a -s --eula accept",
```

Comment:
Does it need to add "bash"? what is it used for?

Reply:
By default, the script is not executable so running it `./{script_path}` requires the extra step of running `chmod +x {script_path}`. But in this case I can probably achieve the same thing by removing back and adding `shell=True`.

Reply:
The `chmod +x` can be set in wget when it's downloaded.